src/parsers/psse_metadata_reimport.jl

```diff
@@ -0,0 +1,160 @@
+#   bus_name_mapping: maps Sienna name to PSS/E name for all buses
+#   bus_number_mapping: maps Sienna number to PSS/E number for all buses
+#   load_name_mapping: maps (Sienna bus, Sienna name) to PSS/E name for all loads
+#   shunt_name_mapping: maps (Sienna bus, Sienna name) to PSS/E name for all shunts
+#   generator_name_mapping: maps (Sienna bus 1, Sienna bus 2, Sienna name) to PSS/E name for all generators
+#   branch_name_mapping: maps (Sienna bus 1, Sienna bus 2, Sienna name) to PSS/E name for all non-transformer branches
+#   transformer_ckt_mapping: maps (Sienna bus1, Sienna bus2, Sienna name) to PSS/E CKT field for all transformers
+
+# Mappings are stored in the file in the Sienna -> PSS/E direction, so they need to be
+# reversed for use here. Wherever a tuple is indicated above, the presentation in the file
+# is of the elements joined with underscores to form a single string.
+
+# NOTE for the moment, this feature is mostly tested in the PowerFlows.jl unit tests.
+
+const PSSE_EXPORT_METADATA_EXTENSION = "_export_metadata.json"
+
+reverse_dict(d::Dict) = Dict(map(reverse, collect(d)))
+
+function split_first_rest(s::AbstractString; delim = "_")
+    splitted = split(s, delim)
+    return first(splitted), join(splitted[2:end], delim)
+end
+
+"Convert a s_bus_n_s_name => p_name dictionary to a (p_bus_n, p_name) => s_name dictionary"
+deserialize_reverse_component_ids(
+    mapping,
+    bus_number_mapping,
+    ::T,
+) where {T <: Type{Int64}} =
+    Dict(
+        let
+            (s_bus_n, s_name) = split_first_rest(s_bus_n_s_name)
+            p_bus_n = bus_number_mapping[s_bus_n]
+            (p_bus_n, p_name) => s_name
+        end
+        for (s_bus_n_s_name, p_name) in mapping)
+deserialize_reverse_component_ids(
+    mapping,
+    bus_number_mapping,
+    ::T,
+) where {T <: Type{Tuple{Int64, Int64}}} =
+    Dict(
+        let
+            (s_buses, s_name) = split_first_rest(s_buses_s_name)
+            (s_bus_1, s_bus_2) = split(s_buses, "-")
+            (p_bus_1, p_bus_2) = bus_number_mapping[s_bus_1], bus_number_mapping[s_bus_2]
+            ((p_bus_1, p_bus_2), p_name) => s_name
+        end
+        for (s_buses_s_name, p_name) in mapping)
+
+"""
+Use PSS/E exporter metadata to build a function that maps component names back to their
+original Sienna values.
+"""
+function name_formatter_from_component_ids(raw_name_mapping, bus_number_mapping, sig)
+    reversed_name_mapping =
+        deserialize_reverse_component_ids(raw_name_mapping, bus_number_mapping, sig)
+    function component_id_formatter(device_dict)
+        (p_bus_n, p_name) = device_dict["source_id"][2:3]
+        (p_bus_n isa Integer) || (p_bus_n = parse(Int64, p_bus_n))
+        new_name = reversed_name_mapping[(p_bus_n, p_name)]
+        return new_name
+    end
+    return component_id_formatter
+end
+
+function remap_bus_numbers!(sys::System, bus_number_mapping)
+    for bus in collect(get_components(Bus, sys))
```

Comment:
why are we calling collect here?

Reply:
because https://github.com/NREL-Sienna/InfrastructureSystems.jl/issues/432